scripting/get5/teamlogic.sp

```diff
@@ -8,11 +8,34 @@ public Action Command_JoinGame(int client, const char[] command, int argc) {
     if (clientTeam == CS_TEAM_NONE || clientTeam == CS_TEAM_SPECTATOR) {
       return Plugin_Continue;
     }
-    ChangeClientTeam(client, clientTeam);
+    // We are required to put a FakeClientCommand on a timer, 
+    // due to ChangeClientTeam having some undefined behaviour
+    // when a coach attempts to call .coach/sm_coach
+    // and have the in-game Valve coaching call not fire off.
+    // In order to avoid this behaviour, we have to use a FakeClientCommand
+    // within a few hundred ms to ensure the client is placed, and can
+    // call .coach on warmup.
+    DataPack playerPack = CreateDataPack();
+    playerPack.WriteCell(GetClientUserId(client));
+    playerPack.WriteCell(clientTeam);
+    CreateTimer(0.1, Timer_PlacePlayerOnJoin, playerPack, TIMER_FLAG_NO_MAPCHANGE);
   }
   return Plugin_Continue;
 }
 
+public Action Timer_PlacePlayerOnJoin(Handle data, DataPack pack) {
+  pack.Reset();
+  int client = GetClientOfUserId(pack.ReadCell());
+  int clientTeam = pack.ReadCell();
+  if (client) {
+    FakeClientCommand(client, "jointeam %d", clientTeam);
+  } else {
+    LogDebug("We are not able to find the client. Refusing to jointeam.");
```

Comment:
Either remove logdebug or make it more clear where it's coming from. I.e. "Player disconnected before they could be automatically placed on a team."